hive表里面如果decimal字段值为空（注意不是NULL），抛出NPE异常

```
Caused by: java.lang.NullPointerException
	at ru.yandex.clickhouse.ClickHousePreparedStatementImpl.setBigDecimal(ClickHousePreparedStatementImpl.java:245)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderBaseTypeStatement(Clickhouse.scala:272)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:298)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:282)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:130)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:122)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

driver(on yarn)的stderr日志最后几行：
```
19/08/08 08:25:58 ERROR TaskSetManager: Task 0 in stage 0.0 failed 4 times; aborting job
19/08/08 08:25:58 INFO ApplicationMaster: Final app status: SUCCEEDED, exitCode: 0
19/08/08 08:25:58 INFO ApplicationMaster: Unregistering ApplicationMaster with SUCCEEDED
19/08/08 08:25:59 INFO ApplicationMaster: Deleting staging directory hdfs://xxx/.sparkStaging/application_1559632632227_39515323
```

driver(on yarn)的stdout相关日志：
```
===============================================================================


Fatal Error, 

Please contact garygaowork@gmail.com or issue a bug in https://github.com/InterestingLab/waterdrop/issues

Reason: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3, bigdata-nmg-hdp3098.nmg01.diditaxi.com, executor 1): java.lang.NullPointerException
	at ru.yandex.clickhouse.ClickHousePreparedStatementImpl.setBigDecimal(ClickHousePreparedStatementImpl.java:245)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderBaseTypeStatement(Clickhouse.scala:272)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:298)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:282)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:130)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:122)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

Driver stacktrace:

Exception StackTrace: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3, bigdata-nmg-hdp3098.nmg01.diditaxi.com, executor 1): java.lang.NullPointerException
	at ru.yandex.clickhouse.ClickHousePreparedStatementImpl.setBigDecimal(ClickHousePreparedStatementImpl.java:245)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderBaseTypeStatement(Clickhouse.scala:272)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:298)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:282)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:130)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:122)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1519)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1507)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1506)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1506)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:814)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:814)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:814)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1734)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1689)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1678)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:630)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2085)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2106)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2125)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2150)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:926)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:364)
	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:926)
	at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply$mcV$sp(Dataset.scala:2345)
	at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply(Dataset.scala:2345)
	at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply(Dataset.scala:2345)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
	at org.apache.spark.sql.Dataset.withNewExecutionId(Dataset.scala:2824)
	at org.apache.spark.sql.Dataset.foreachPartition(Dataset.scala:2344)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.process(Clickhouse.scala:122)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$batchProcessing$2.apply(Waterdrop.scala:205)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$batchProcessing$2.apply(Waterdrop.scala:204)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at io.github.interestinglab.waterdrop.Waterdrop$.batchProcessing(Waterdrop.scala:204)
	at io.github.interestinglab.waterdrop.Waterdrop$.io$github$interestinglab$waterdrop$Waterdrop$$entrypoint(Waterdrop.scala:114)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$1.apply$mcV$sp(Waterdrop.scala:35)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$1.apply(Waterdrop.scala:35)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$1.apply(Waterdrop.scala:35)
	at scala.util.Try$.apply(Try.scala:192)
	at io.github.interestinglab.waterdrop.Waterdrop$.main(Waterdrop.scala:35)
	at io.github.interestinglab.waterdrop.Waterdrop.main(Waterdrop.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.yarn.ApplicationMaster$$anon$2.run(ApplicationMaster.scala:675)
Caused by: java.lang.NullPointerException
	at ru.yandex.clickhouse.ClickHousePreparedStatementImpl.setBigDecimal(ClickHousePreparedStatementImpl.java:245)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderBaseTypeStatement(Clickhouse.scala:272)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:298)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:282)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:130)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:122)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


===============================================================================

```

```
===============================================================================


Fatal Error, 

Please contact garygaowork@gmail.com or issue a bug in https://github.com/InterestingLab/waterdrop/issues

Reason: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3, bigdata-nmg-hdp3198.nmg01.diditaxi.com, executor 1): java.util.NoSuchElementException: key not found: pt
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:296)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:294)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:142)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:134)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

Driver stacktrace:

Exception StackTrace: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3, bigdata-nmg-hdp3198.nmg01.diditaxi.com, executor 1): java.util.NoSuchElementException: key not found: pt
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:296)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:294)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:142)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:134)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1519)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1507)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1506)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1506)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:814)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:814)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:814)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1734)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1689)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1678)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:630)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2085)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2106)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2125)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2150)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:926)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:364)
	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:926)
	at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply$mcV$sp(Dataset.scala:2345)
	at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply(Dataset.scala:2345)
	at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply(Dataset.scala:2345)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
	at org.apache.spark.sql.Dataset.withNewExecutionId(Dataset.scala:2824)
	at org.apache.spark.sql.Dataset.foreachPartition(Dataset.scala:2344)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.process(Clickhouse.scala:134)
	at io.github.interestinglab.waterdrop.Waterdrop$.outputProcess(Waterdrop.scala:250)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$batchProcessing$2.apply(Waterdrop.scala:216)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$batchProcessing$2.apply(Waterdrop.scala:215)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at io.github.interestinglab.waterdrop.Waterdrop$.batchProcessing(Waterdrop.scala:215)
	at io.github.interestinglab.waterdrop.Waterdrop$.io$github$interestinglab$waterdrop$Waterdrop$$entrypoint(Waterdrop.scala:116)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$1.apply$mcV$sp(Waterdrop.scala:37)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$1.apply(Waterdrop.scala:37)
	at io.github.interestinglab.waterdrop.Waterdrop$$anonfun$1.apply(Waterdrop.scala:37)
	at scala.util.Try$.apply(Try.scala:192)
	at io.github.interestinglab.waterdrop.Waterdrop$.main(Waterdrop.scala:37)
	at io.github.interestinglab.waterdrop.Waterdrop.main(Waterdrop.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.yarn.ApplicationMaster$$anon$2.run(ApplicationMaster.scala:675)
Caused by: java.util.NoSuchElementException: key not found: pt
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement$1.apply$mcVI$sp(Clickhouse.scala:296)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse.io$github$interestinglab$waterdrop$output$batch$Clickhouse$$renderStatement(Clickhouse.scala:294)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:142)
	at io.github.interestinglab.waterdrop.output.batch.Clickhouse$$anonfun$process$1.apply(Clickhouse.scala:134)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:928)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2125)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


===============================================================================
```

--- a/waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/Waterdrop.scala
+++ b/waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/Waterdrop.scala
@@ -39,7 +39,7 @@ object Waterdrop extends Logging {
               case Failure(exception) => {
                 exception match {
                   case e @ (_: ConfigRuntimeException | _: UserRuntimeException) => Waterdrop.showConfigError(e)
-                  case e: Exception => showFatalError(e)
+                  case e: Exception => throw new Exception(e)
                 }
               }
             }
@@ -75,9 +75,11 @@ object Waterdrop extends Logging {
     println("Config Error:\n")
     println("Reason: " + errorMsg + "\n")
     println("\n===============================================================================\n\n\n")
+    throw new ConfigRuntimeException(throwable)
   }
 
   private[waterdrop] def showFatalError(throwable: Throwable): Unit = {
+
     println("\n\n===============================================================================\n\n")
     val errorMsg = throwable.getMessage
     println("Fatal Error, \n")
@@ -86,6 +88,7 @@ object Waterdrop extends Logging {
     println("Reason: " + errorMsg + "\n")
     println("Exception StackTrace: " + ExceptionUtils.getStackTrace(throwable))
     println("\n===============================================================================\n\n\n")
+    throw new Exception(throwable)
   }
 
   private def entrypoint(configFile: String): Unit = {

--- a/waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/config/ConfigBuilder.scala
+++ b/waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/config/ConfigBuilder.scala
@@ -25,16 +25,21 @@ class ConfigBuilder(configFile: String) {
     println("[INFO] Loading config file: " + configFile)
 
     // variables substitution / variables resolution order:
-    // onfig file --> syste environment --> java properties
-    val config = ConfigFactory
-      .parseFile(new File(configFile))
-      .resolve(ConfigResolveOptions.defaults().setAllowUnresolved(true))
-      .resolveWith(ConfigFactory.systemProperties, ConfigResolveOptions.defaults.setAllowUnresolved(true))
+    // config file --> syste environment --> java properties
 
-    val options: ConfigRenderOptions = ConfigRenderOptions.concise.setFormatted(true)
-    println("[INFO] parsed config file: " + config.root().render(options))
+    try {
+      val config = ConfigFactory
+        .parseFile(new File(configFile))
+        .resolve(ConfigResolveOptions.defaults().setAllowUnresolved(true))
+        .resolveWith(ConfigFactory.systemProperties, ConfigResolveOptions.defaults.setAllowUnresolved(true))
 
-    config
+      val options: ConfigRenderOptions = ConfigRenderOptions.concise.setFormatted(true)
+      println("[INFO] parsed config file: " + config.root().render(options))
+
+      config
+    } catch {
+      case e: Throwable => throw new ConfigRuntimeException(e)
+    }
   }
 
   /**

--- a/waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/config/ConfigBuilder.scala
+++ b/waterdrop-core/src/main/scala/io/github/interestinglab/waterdrop/config/ConfigBuilder.scala
@@ -9,6 +9,7 @@ import scala.collection.JavaConverters._
 import com.typesafe.config.{Config, ConfigFactory, ConfigRenderOptions, ConfigResolveOptions}
 import io.github.interestinglab.waterdrop.apis._
 
+import scala.util.{Failure, Success, Try}
 import util.control.Breaks._
 
 class ConfigBuilder(configFile: String) {
@@ -27,7 +28,7 @@ class ConfigBuilder(configFile: String) {
     // variables substitution / variables resolution order:
     // config file --> syste environment --> java properties
 
-    try {
+    Try({
       val config = ConfigFactory
         .parseFile(new File(configFile))
         .resolve(ConfigResolveOptions.defaults().setAllowUnresolved(true))
@@ -37,8 +38,9 @@ class ConfigBuilder(configFile: String) {
       println("[INFO] parsed config file: " + config.root().render(options))
 
       config
-    } catch {
-      case e: Throwable => throw new ConfigRuntimeException(e)
+    }) match {
+      case Success(conf) => conf
+      case Failure(exception) => throw new ConfigRuntimeException(exception)
     }
   }
 

